libs/commons/Testutil_files.mli

```diff
@@ -0,0 +1,63 @@
+(*
+   Utilities for creating, scanning, and deleting a hierarchy
+   of test files.
+*)
+
+(*
+   We don't deal with other file types than those three.
+   File names must be normal names, i.e. not ".", "..", or "". They may not
+   contain slashes or backslashes either.
+*)
+type t =
+  | Dir of string * t list
+  | File of string * string
+  | Symlink of string * string
+
+(*
+   Sort the files in a reasonable order. Useful for comparison purposes.
+*)
+val sort : t list -> t list
+
+(*
+   Return one path per file or symlink the tree.
+   'include_dirs=true' will return the path to the folders as well.
+*)
+val flatten : ?root:Fpath.t -> ?include_dirs:bool -> t list -> Fpath.t list
+
+(*
+   Read files from a root. Don't follow symlinks.
+   Fail with an exception if we can't read the files or if they're of an
+   exotic kind.
+*)
+val read : Fpath.t -> t list
+
+(*
+   Write a file hierarchy into the given root folder,
+   which must already exist.
+*)
+val write : Fpath.t -> t list -> unit
+
+(* Recursive removal (rm -r) *)
+val remove : Fpath.t -> unit
+val is_dir : Fpath.t -> bool
```

Comment:
Fpath does not provide such functions already?

Reply:
No, because Fpath doesn't access the file system.